File: demo/domain/produit.c
#include "produit.h"
#include <stdio.h>

const char* getTypeProduitString(TypeProduit type) {
    switch (type) {
        case VOITURE: return "Voiture";
        case MAISON: return "Maison";
        case MOTO: return "ASSIETTE";
        default: return "Inconnu";
    }
}

void afficherProduit(const Produit* p) {
    printf("%d. %s - %d€ - %s\n", p->id, getTypeProduitString(p->type), p->prix,
           p->etat == DISPONIBLE ? "Disponible" : "Indisponible");
}

File: demo/domain/Produit.js
class Produit {
  constructor(id, type, prix) {
    this.id = id;
    this.type = type;
    this.prix = prix;
    this.disponible = true;
  }

  afficher() {
    console.log(
      `${this.id}. ${this.type} - ${this.prix}€ - ${this.disponible ? "Disponible" : "Indisponible"}`
    );
  }
}

module.exports = Produit;
